### PR TITLE
Entity parent child pom

### PR DESCRIPTION
**Pull request by BogdanQA**
https://trello.com/c/cAqKNaYB

Refactoring ParentChildTest 

EntityParentTest
EntityChildTest
EntityParentChildViewTest

Merge all together in POM

### CODE REVIEW

**Review by guram21 — APPROVED** [review]

**Review by Burdishka — APPROVED** [review]

File: src/test/java/model/entity/view/ParentViewPage.java [hunk 1]
```diff
@@ -17,30 +20,60 @@
...
+    private WebElement inputDate;
+
+    @FindBy(xpath = "//input[contains(@name, 'datetime')]")
+    private WebElement inputDateTime;
+
+    @FindBy(xpath = "//button[@data-id='user']/div/div")
+    private WebElement inputUser;
+
+    @FindBy(xpath = ("//button[text() = 'Save']"))
+    private WebElement buttonSave;
```
Comment (MaxBurdin) [inline src/test/java/model/entity/view/ParentViewPage.java:54]: These elements belong to edit page. Not view page

File: src/test/java/model/entity/view/ParentViewPage.java [hunk 2]
```diff
@@ -17,30 +20,60 @@
...
 
     public String viewDate() {
         return tab.findElement(By.xpath("//label[text()='String']/../div[5]//span")).getText();
     }
 
+    public String viewDateTime() { return tab.findElement(By.xpath("//label[text()='String']/../div[6]//span")).getText(); }
+
     public String viewUser(){
             return tab.findElement(By.xpath("//div[@class='row']//p")).getText();
     }
```
Comment (MaxBurdin) [inline src/test/java/model/entity/view/ParentViewPage.java:79]: You don't need all these methods here. There are labeledValues element and getValues() method in EntityBaseViewPage that you can use for getting values and verifying in your test.

File: src/test/java/test/entity/EntityParentChildTest.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                .deleteRow();
+
+        RecycleBinPage recycleBinPage = ParentTablePage.clickRecycleBin();
+
+        Assert.assertEquals(ParentTablePage.getRowCount(), 0);
+        Assert.assertEquals(recycleBinPage.getRowCount(), 1);
+    }
+
+    @Test(dependsOnMethods = "deleteParentTest")
+    public void createMultipleParent() {
```
Comment (MaxBurdin) [inline src/test/java/test/entity/EntityParentChildTest.java:170]: Each test must include the word Test in its name

File: src/test/java/test/entity/EntityParentChildTest.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                .clickSaveButton()
+                .clickNewFolder()
+                .sendKeys2(DRUG_THE_ROW_DOWN)
+                .clickSaveButton();
+
+        Assert.assertEquals(ParentTablePage.getRowCount(), 2);
+    }
+
+    @Test(dependsOnMethods = "createMultipleParent")
+    public void dragTheRowUp() {
```
Comment (MaxBurdin) [inline src/test/java/test/entity/EntityParentChildTest.java:185]: Each test must include the word Test in its name

File: src/test/java/test/portal/UserCreationByAdminDraftTest.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+public class UserCreationByAdminDraftTest extends BaseTest {
+
+    private static final String EMAIL = "test@test.com";
+    private static final String FULLNAME = "Test Test";
+    private static final String ROLE = "Test Role";
+    private static final String PERSONAID = "Test Persona";
+
+    @Test
+    public void newUser() {
```
Comment (MaxBurdin) [inline src/test/java/test/portal/UserCreationByAdminDraftTest.java:18]: Each test must include word Test in its name

File: src/test/java/test/portal/UserCreationByAdminDraftTest.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        WebDriver driver = getDriver();
+
+        WebElement users = driver.findElement(By.xpath("//p[contains(text(),'Users')]"));
+        TestUtils.click(driver, users);
+
```
Comment (MaxBurdin) [inline src/test/java/test/portal/UserCreationByAdminDraftTest.java:23]: Empty line. And this test missing assertion

**Review by MaxBurdin — CHANGES_REQUESTED** [review]

File: src/test/java/model/base/BasePage.java [hunk 7]
```diff
@@ -1,5 +1,6 @@
 package model.base;
 
+import model.entity.edit.ParentEditPage;
```
Comment (MaxBurdin) [inline src/test/java/model/base/BasePage.java:3]: This import is unnecessary here

File: src/test/java/model/entity/table/ParentTablePage.java [hunk 8]
```diff
@@ -38,26 +38,30 @@ protected ParentViewPage createViewPage() {
...
     }
 
-    public String getDataText(){
+    public String getDate(){
         return table.findElement(By.xpath("//tr/td[6]/a")).getText();
     }
 
+    public String getDateTime(){
+        return table.findElement(By.xpath("//tr/td[7]/a")).getText();
+    }
```
Comment (MaxBurdin) [inline src/test/java/model/entity/table/ParentTablePage.java:63]: You don't need all these methods here. There is getRow() method in EntityBasePage that you can use for getting values and verifying in your test.

**Review by MaxBurdin — CHANGES_REQUESTED** [review]

File: src/test/java/model/base/EntityBaseViewPage.java [hunk 9]
```diff
@@ -9,7 +9,7 @@
 import java.util.List;
 import java.util.stream.Collectors;
 
-public class EntityBaseViewPage extends BasePage {
+public class EntityBaseViewPage extends BaseIndexPage {
```
Comment (MaxBurdin) [inline src/test/java/model/base/EntityBaseViewPage.java:12]: Why did you decide to inherit from BaseIndexPage?

Comment (oooleks) [reply src/test/java/model/base/EntityBaseViewPage.java:12]: This makes sense to extend from BaseIndexPage as BaseIndexPage contains elements and methods like: getCurrentUser(); resetUserData(); logout() you will not be able to call from view page without such an extension.

File: src/test/java/test/portal/UserCreationByAdminDraftTest.java (hunk 5 above)
Comment (oooleks) [reply src/test/java/test/portal/UserCreationByAdminDraftTest.java:18]: Try to create test name with the template like: **verb** (action on subject ("create", "edit", "view" etc)) + **noun** (subject under test) + word "**Test**"

File: src/test/java/model/entity/view/ParentViewPage.java (hunk 1 above)
Comment (BogdanQA) [reply src/test/java/model/entity/view/ParentViewPage.java:54]: All this methods are for Child menu, which is on Parent view page

File: src/test/java/test/entity/EntityParentChildTest.java (hunk 3 above)
Comment (BogdanQA) [reply src/test/java/test/entity/EntityParentChildTest.java:170]: Fixed

File: src/test/java/test/entity/EntityParentChildTest.java (hunk 4 above)
Comment (BogdanQA) [reply src/test/java/test/entity/EntityParentChildTest.java:185]: Fixed

File: src/test/java/test/portal/UserCreationByAdminDraftTest.java (hunk 5 above)
Comment (BogdanQA) [reply src/test/java/test/portal/UserCreationByAdminDraftTest.java:18]: This commit here by mistake, Fixed

File: src/test/java/test/portal/UserCreationByAdminDraftTest.java (hunk 6 above)
Comment (BogdanQA) [reply src/test/java/test/portal/UserCreationByAdminDraftTest.java:23]: This commit here by mistake, Fixed

File: src/test/java/model/base/BasePage.java (hunk 7 above)
Comment (BogdanQA) [reply src/test/java/model/base/BasePage.java]: Fixed

File: src/test/java/model/entity/view/ParentViewPage.java (hunk 1 above)
Comment (MaxBurdin) [reply src/test/java/model/entity/view/ParentViewPage.java:54]: Do you mean these elements become available when you're adding a child record? Then you need to create a child edit page and inherit it from EntityBaseEditExtPage which has the fields. But what do you need them for anyway? They are not used

**Review by oooleks — APPROVED** [review]

**Review by MaxBurdin — APPROVED** [review]

**Pull request merged**